dvc/external_repo.py

```diff
@@ -19,7 +21,10 @@ def external_repo(url=None, rev=None, rev_lock=None, cache_dir=None):
 
     path = _external_repo(url=url, rev=rev_lock or rev, cache_dir=cache_dir)
     repo = Repo(path)
-    yield repo
+    try:
+        yield repo
+    except OutputNotFoundError as cause:
+        raise NoOutputInExternalRepoError(url, cause.failed_output)
```

Comment:
I know I suggested it, but it is fragile. We don't know whether that `OutputNotFoundError` we catch here comes from the yielded `repo`. To fix this we will probably need to pass repo to `OutputNotFoundError` and check it here:
```python
if cause.repo is repo:
    # wrap
else:
    raise
```
I also wouldn't call it `cause` but simply `exc`, it only becomes a cause when we wrap it not when we catch it.